R/nbinomdm.R

```diff
@@ -0,0 +1,79 @@
+#' Negative Binomial Data Maker
+#'
+#' nbinomdm uses rnbinom function from the stats package to create a data frame
+#' containing different samples for different treatment groups following the
+#' negative binomial distribution. Currently it uses the size and mu parameters
+#' following the rnbinom function, but not the prob parameter.
+#'
+#' @param k A single value corresponding to the number of treatment groups.
+#' @param names A vector of character strings specifying the names of the treatment groups.
+#' @param n A vector specifying the sample sizes of the treatment groups.
+#' @param size A vector specifying the dispersion parameters of the negative binomial distributions for the treatment groups. Strictly have to be positive values, not necessarily integers.
+#' @param mu A vector specifying the means of the negative binomial distributions for the treatment groups.
+#'
+#' @return Histogram and jitter plots for the treatment groups and a data frame containing the random samples.
+#' @export
+#' @importFrom rlang .data
+#'
+#' @examples
+#' nbinomdm(k=3,names=c("Control","Treatment1","Treatment2"),n=c(50,50,60),size=c(3,3,3),mu=c(5,10,20))
+#'
+#'
+
+nbinomdm <- function(k,names,n,size,mu){
+  
+  if(length(n) != k
+     | length(size) != k
+     | length(mu) != k
+     | length(names) != k
+  )
+  {
+    print("Length of sample size or size or prob doesn't match the number of groups")
+    stop()
+  }
+  
+  list_of_values <- lapply(1:k,
+                           FUN = function(c){
+                             stats::rnbinom(
+                               n=n[c]
+                               ,size=size[c]
+                               ,mu=mu[c]
+                             )
+                           }
+  )
+  
+  Data <- as.data.frame(t(
+    purrr::map_dfr(.x = list_of_values, .f = ~ as.data.frame(t(.)))
+  ))
+  colnames(Data) <- names
+  
+  Input_parameters <- data.frame(groups = names
+                                 , n = n
+                                 , size = size
+                                 , mu = mu)
+  
+  
+  DataFrame_p <- tidyr::pivot_longer(data = Data
+                                     ,cols = dplyr::everything()
+                                     ,names_to = "Groups"
+                                     ,values_to = "Value")
+  DataFrame_p <- stats::na.omit(DataFrame_p)
+  
+  p1 <- ggplot2::ggplot(DataFrame_p,
+                        ggplot2::aes(x=.data$Value,fill = .data$Groups)) +
+    ggplot2::geom_histogram(bins = 15,position = "identity", alpha = .8) +
+    ggplot2::theme_bw()
+  p2 <- ggplot2::ggplot(DataFrame_p,
+                        ggplot2::aes(x=.data$Groups,y = .data$Value,color=.data$Groups)) +
+    ggplot2::geom_jitter() +
+    ggplot2::theme_bw()
+  
+  print(p1)
+  print(p2)
+  print(Input_parameters)
+  
+  return(Data)
```

Comment:
how about if we return a list object? It has 1) Input_parameters, 2) Data, 3) p1 (optional), 4) p2 (optional), 5) Dataframe_p

Again, the Dataframe_p is what is fed into the statistical test, and this test and pvalue collection is replicated inside of the repeater functions.

The reason the plots are optional is that I'm worried they take too much clock time. I don't think they need to be made one thousand times inside of the repeater.